src/python/zquantum/qcbm/cost_function.py

```diff
@@ -0,0 +1,114 @@
+from zquantum.core.interfaces.cost_function import CostFunction
+from zquantum.core.interfaces.backend import QuantumBackend
+from zquantum.core.bitstring_distribution import BitstringDistribution, evaluate_distribution_distance
+from zquantum.core.circuit import build_ansatz_circuit
+from typing import Union, Dict
+import numpy as np
+
+class QCBMCostFunction(CostFunction):
+    """
+    Cost function used for evaluating QCBM.
+
+    Args:
+        ansatz (dict): dictionary representing the ansatz
+        backend (zquantu.core.interfaces.backend.QuantumBackend): backend used for QCBM evaluation
+        distance_measure (str): string describing which distance measure should be used. See zquantum.core.bitstring_distribution.evaluate_distribution_distance
+        target_bitstring_distribution (zquantum.core.bitstring_distribution.BitstringDistribution): bistring distribution which QCBM aims to learn
+        epsilon (float): clipping value used for calculating distribution distance
```

Comment:
I think epsilon should be like an optional arg/we should use kwargs, because this parameter makes sense only if we use a certain distance measure. I think I'd implemented it like that at the very beginning, and then maybe @simonwa7 made it simpler since we have only one distance measure implemented?

Also, I don't see why we want to have a specific cost function class for qcbm? Why isn't there only one cost function which works for whatever ansatz? @mstechly you probably had some good reasons for this, looking forward to hearing :)

Reply:
Ad `epsilon` – I think I would leave it as it is for now. Refactoring it right now might a little bit "blind" and premature. Once we actually will need another distance measure, we'll have much better idea how to refactor it so that it makes sense. 

Ad specific function – the main reason is that this one uses `get_bitstring_distribution` and not `get_expectation_value`. But this might be actually a good point and we might want to merge it with the `EvaluateOperatorCostFunction` function at some point.

Reply:
Agree for epsilon 👍
I think the second point is important, and it comes up because we have implemented both VQE and QML which are slightly different as for the context they live in, event though they are both VQAs. Would love to figure out a "unified" implementation, could be helpful for users to better navigate the platform eventually! How about writing a story for this, since there's also the question on how to write cost functions for ML?